### PR TITLE
backfill storage ids

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Follow on to https://github.com/code-dot-org/code-dot-org/pull/17905, or part 3 in this list

1. [migration] Create new (indexed) column in channel_tokens called storage_id
2. [code change] Update code to write both user_id and storage_id columns for new rows
3. [migration] Backfill storage_id, derived by decrypting channel column. On production this will be managed via a script that batches updates. In other envs this will involve a migration.
4. [migration] Make storage_id non-nullable, make user_id/channel nullable (now that all rows should be backfilled and new rows are getting storage_id from 2
5. [code change] Stop using channel/user_id, use storage_id instead
6. [migration] Delete channel/user_id columns

Now that new rows are getting a storage_id, we can start backfilling old rows. For databases where we have less than 100k rows, we might as well just do this as part of the migration. For larger databases (ie. prod), I will instead run a one-off script which can be managed (and doesnt need to happen all at once).

### CODE REVIEW

File: dashboard/db/migrate/20170923000355_populate_storage_id.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class PopulateStorageId < ActiveRecord::Migration[5.0]
+  def change
+    # We don't want to run as part of our deploy if the table is large (i.e. prod)
+    # Prod will instead do this using the very similar oneoff script
+    # bin/oneoff/backfill_data/channel_tokens_storage_id
+    return if ChannelToken.count > 100000
```
Comment (Bjvanminnen) [inline dashboard/db/migrate/20170923000355_populate_storage_id.rb]: Not sure if this is the best way to accomplish this. Could instead check if env != production.

Asher's approach for an equivalent change seemed to be to get devs to run this manually https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/db/migrate/20170818182341_required_storage_app_id_in_channel_tokens.rb

**Comment by Bjvanminnen** [comment]
Script code was based off of https://github.com/code-dot-org/code-dot-org/blob/staging/bin/oneoff/backfill_data/channel_tokens_storage_app_id

File: dashboard/db/migrate/20170923000355_populate_storage_id.rb (hunk 1 above)
Comment (jopolsky) [reply dashboard/db/migrate/20170923000355_populate_storage_id.rb]: I prefer the check to env instead.

Comment (davidsbailey) [reply dashboard/db/migrate/20170923000355_populate_storage_id.rb]: +1 for checking env.

> Asher's approach for an equivalent change seemed to be to get devs to run this manually

I like what you have here better. This should happen automatically to developers unless there's a reason that won't work.

Comment (Bjvanminnen) [reply dashboard/db/migrate/20170923000355_populate_storage_id.rb]: I'm surprised not to find any existing migrations where we check environment, though I think it should be as simple as `Rails.env.production?`.

**Comment by Bjvanminnen** [comment]
"Is there another approach in ruby to share the logic that runs the backfill?" There are certainly ways we could share it, but I'm not sure we have any good precedents for how to. Because the production version is meant to be a one-off, I don't feel too badly about duplicating code.

**Comment by Bjvanminnen** [comment]
For posterity, this took a bit over 5 hours to run on prod

**Pull request merged**